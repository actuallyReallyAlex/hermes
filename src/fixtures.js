export const defaultState = {
  ship: {
    cargo: {
      items: [],
      volumeRemaining: 5
    },
    location: {
      name: null,
      value: null
    }
  },
  ui: {
    view: 'Ship'
  },
  user: {
    cash: 100
  },
  world: {
    isTimerRunning: false,
    planets: [
      {
        id: "0",
        isHomePlanet: true,
        location: 0,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '0',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '1',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '2',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '3',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '4',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 1'
      },
      {
        id: "1",
        isHomePlanet: false,
        location: 50,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '5',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '6',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '7',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '8',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '9',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          }
        ],
        name: 'Test Planet 2'
      },
      {
        id: "2",
        isHomePlanet: false,
        location: 100,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '10',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '11',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '12',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '13',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '14',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 3'
      }
    ]
  }
}
