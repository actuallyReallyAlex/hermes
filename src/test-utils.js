import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { defaultState } from './fixtures'
import { render } from '@testing-library/react'

const mockThunks = ({ dispatch, getState }) => next => action => {
  const mockedAction = () => ({ type: 'MOCKED ACTION' })

  let returnValue

  if (typeof action === 'function' || action === undefined) {
    returnValue = mockedAction
  } else {
    returnValue = next(action)
  }

  return returnValue
}

const middlewares = [mockThunks]

export const mockStore = configureStore(middlewares)

export const customRender = options => {
  const { component, state, props } = options

  const Component = component

  return render(
    <Provider store={mockStore(state ? state : defaultState)}>
      <Component {...props} />
    </Provider>
  )
}
