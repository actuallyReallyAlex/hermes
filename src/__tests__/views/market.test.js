import { customRender } from '../../test-utils'
import { defaultState } from '../../fixtures'
import MarketView from '../../views/market'

const customState = {
  ...defaultState,
  ui: {
    isCreatingContract: false,
    view: 'Market'
  },
  ship: {
    ...defaultState.ship,
    location: { name: 'Test Planet 1', value: 0 },
    cargo: {
      items: [
        {
          destination: {
            name: 'Test Planet 2',
            value: 50
          },
          id: '0',
          name: 'Test Item',
          quantity: 1,
          value: 1,
          volume: 1
        }
      ],
      volumeRemaining: 4
    }
  }
}

describe('<MarketView />', () => {
  it('Should render the <MarketView /> component.', () => {
    const container = customRender({
      component: MarketView,
      state: customState
    })
    expect(container.asFragment()).toMatchSnapshot()
  })
})
