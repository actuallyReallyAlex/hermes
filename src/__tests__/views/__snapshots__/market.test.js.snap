// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarketView /> Should render the <MarketView /> component. 1`] = `
<DocumentFragment>
  <div
    class="StyledBox-sc-13pk1d4-0 fWrrBh"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <div
        class="StyledTabs-a4fwxl-2 jBsJwq StyledBox-sc-13pk1d4-0 fWrrBh"
        role="tablist"
      >
        <div
          class="StyledTabs__StyledTabsHeader-a4fwxl-0 cjlqAH StyledBox-sc-13pk1d4-0 ckzykU"
        >
          <button
            aria-expanded="true"
            aria-selected="true"
            class="StyledButton-sc-323bzc-0 jfbKsv"
            role="tab"
            type="button"
          >
            <div
              class="StyledTab-sc-1nnwnsb-0 dGMZCN StyledBox-sc-13pk1d4-0 hAQvDd"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 bWRWBf"
              >
                <svg
                  aria-label="LinkDown"
                  class="StyledIcon-ofa7kd-0 iOkQrb"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,2 L12,22 M3,11 L12,2 L21,11"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                    transform="matrix(1 0 0 -1 0 24)"
                  />
                </svg>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gufrXq"
                />
                <span
                  class="StyledText-sc-1sadyjn-0 gNZQxf"
                >
                  <strong>
                    Buy
                  </strong>
                </span>
              </div>
            </div>
          </button>
          <button
            aria-expanded="false"
            aria-selected="false"
            class="StyledButton-sc-323bzc-0 jfbKsv"
            role="tab"
            type="button"
          >
            <div
              class="StyledTab-sc-1nnwnsb-0 dGMZCN StyledBox-sc-13pk1d4-0 kyubVG"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 bWRWBf"
              >
                <svg
                  aria-label="LinkUp"
                  class="StyledIcon-ofa7kd-0 iOkQrb"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,2 L12,22 M3,11 L12,2 L21,11"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gufrXq"
                />
                <span
                  class="StyledText-sc-1sadyjn-0 gNZQxf"
                >
                  <strong>
                    Sell
                  </strong>
                </span>
              </div>
            </div>
          </button>
        </div>
        <div
          aria-label="1 Tab Contents"
          class="StyledTabs__StyledTabPanel-a4fwxl-1 lchlbl"
          role="tabpanel"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 ekjAwv"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
              style="width: 200px;"
            >
              <ul
                class="MuiList-root MuiList-padding"
                role="menu"
                tabindex="-1"
              >
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root Mui-selected MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected"
                  role="menuitem"
                  tabindex="-1"
                >
                  Ore
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Plasma
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Space Junk
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Antibiotics
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Tobacco
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Soil
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Ammunition
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Glass
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Silver
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
                <li
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Gold
                  <span
                    class="MuiTouchRipple-root"
                  />
                </li>
              </ul>
            </div>
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
              style="width: 100%;"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <div
                    class="MuiBox-root MuiBox-root-53"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-54"
                    >
                      <h2
                        class="MuiTypography-root MuiTypography-h5"
                      >
                        Ore
                      </h2>
                      <svg
                        aria-label="Cloud"
                        class="StyledIcon-ofa7kd-0 iOkQrb"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18,17 L18,18 C18,21 16,22 13,22 L11,22 C8,22 6,21 6,18 L6,17 C3.23857625,17 1,14.7614237 1,12 C1,9.23857625 3.23857625,7 6,7 L12,7 M6,7 L6,6 C6,3 8,2 11,2 L13,2 C16,2 18,3 18,6 L18,7 C20.7614237,7 23,9.23857625 23,12 C23,14.7614237 20.7614237,17 18,17 L12,17"
                          fill="none"
                          stroke="#000"
                          stroke-width="2"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-85"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                      >
                        Your garden variety ore. From a rock. A space rock.
                      </p>
                      <span
                        class="MuiTypography-root MuiTypography-caption"
                      >
                        Value: 2
                      </span>
                      <br />
                      <span
                        class="MuiTypography-root MuiTypography-caption"
                      >
                        Volume: 1
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <table
                class="MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head"
                      scope="col"
                    >
                      Name
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-head"
                      scope="col"
                    >
                      Price
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-head"
                      scope="col"
                    >
                      Location
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Sergio
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥9.87
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Argentina
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Vonny
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥6.28
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Russia
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Cassy
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥8.87
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Poland
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Isabelita
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥1.23
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Nicaragua
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Georas
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥9.90
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Indonesia
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Milly
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥2.44
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Sweden
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Gene
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥7.23
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Poland
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Liliane
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥5.48
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Philippines
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Roi
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥2.12
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      Portugal
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                    >
                      Casper
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      ¥1.01
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
                    >
                      France
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
